src/request.ts

```diff
@@ -336,13 +339,24 @@ class DatastoreRequest {
    *   const apiResponse = data[0];
    * });
    */
-  delete(keys, gaxOptions?, callback?) {
-    if (is.fn(gaxOptions)) {
-      callback = gaxOptions;
-      gaxOptions = {};
-    }
-
-    callback = callback || (() => {});
+  delete(keys: Entities):
+      void|Promise<google.datastore.v1.Datastore.CommitCallback>;
```

Comment:
No need to explicitly return `void` here

Reply:
Agreed, but I added the void to solve this error:
```
Property 'delete' in type 'Transaction' is not assignable to the same property in base type 'DatastoreRequest'.
  Type '(entities: any) => void' is not assignable to type '{ (keys: any): Promise<CommitCallback>; (keys: any, callback: CommitCallback): void; (keys: any, gaxOptions: CallOptions, callback: CommitCallback): void; }'.
    Type 'void' is not assignable to type 'Promise<CommitCallback>'.
```
delete in transaction.ts has a void return annotation, and when changed to the same Promise<CommitCallback> there's an error asking explicitly for an any or void return annotation. I kept the promise to follow the "no callback = return promise" rule.

Should I make them all void and remove the promise return annotation from the method definition? (if all method overloads have the same return annotation, do we really need the overloads?)

Reply:
Well hold on nah - what does `CommitCallback` have in it's type?  Is it a callback function type?  The answer here may be to return `Promise<void>`.

Reply:
Yeah, `CommitCallback` is a callback function type. I also tried `Promise<void>` but I'm getting the same errors. I'll ask around to see if we can come up with something using feedback from an IDE

Reply:
The error I'm solving for:
```
Type 'void' must have a '[Symbol.iterator]()' method that returnsan iterator.

826       const [[deletedEntity], [fetchedEntity]] = await Promise.all([
```